src/fcp/result.py

```diff
@@ -1,71 +1,55 @@
 import sys
 import traceback
 
+from typing import Any, Union
 from functools import wraps
 
 
-class ResultShortcutError(Exception):
-    def __init__(self, error):
-        super().__init__()
-        self.error = error
-
-
-def result_shortcut(f):
-    @wraps(f)
-    def wrapper(*args, **kwargs):
-        try:
-            return f(*args, **kwargs)
-        except ResultShortcutError as err:
-            return err.error
-
-    return wrapper
-
-
 class Result:
-    def is_err(self):
+    def is_err(self) -> bool:
         return isinstance(self, Error)
 
-    def is_ok(self):
+    def is_ok(self) -> bool:
         return isinstance(self, Ok)
 
 
 class Ok(Result):
-    def __init__(self, value):
+    def __init__(self, value: Any) -> None:
         self.value = value
 
-    def unwrap(self):
+    def unwrap(self) -> Any:
         return self.value
 
-    def err(self):
+    def err(self) -> None:
         sys.exit(1)
 
-    def compound(self, result):
+    def compound(self, result: Any) -> Any:
         if isinstance(result, Ok):
             v1 = self.value if isinstance(self.value, list) else [self.value]
             v2 = result.value if isinstance(result.value, list) else [result.value]
             return Ok(v1 + v2)
         else:
             return result
 
-    def Q(self):
+    def Q(self) -> Any:
         return self.value
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
         if not isinstance(other, Ok):
             return False
 
-        return self.value == other.value
+        return bool(self.value == other.value)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "Ok"
 
 
 class Error(Result):
-    def __init__(self, error):
-        self.error = error
+    def __init__(self, error: Union[list[str], str]) -> None:
```

Comment:
```suggestion
    def __init__(self, error: Any) -> None:
```

?

I would expect that error value could be anything, and not restricted to strings or lists of strings

Reply:
The way the __repr__ method is implemented will only work with lists of strings. I can make a change to Any but it seams the class was designed to receive and list of str representing an error stack trace.

Reply:
__repr__ will work with any type. It does have special handling for lists.

I guess that typing that may be hard.